Sources/SwiftParser/Lexer.swift

```diff
@@ -150,6 +154,10 @@ extension Lexer {
     func peek() -> Lexer.Lexeme {
       return self.nextToken
     }
+
+    public var debugDescription: String {
+      return self.nextToken.debugDescription + String(syntaxText: SyntaxText(baseAddress: self.cursor.input.baseAddress, count: self.cursor.input.count))
```

Comment:
This is backwards - it’ll print the next then the current source text behind it. If there were delimiters here that call that out I’d be happier.

Reply:
My ide was that `po self.lexemes` should be analogous to `po *Ptr` (or something like this) in the C++ parser, i.e. print the remaining source text, except for the current token. And AFAICT that’s what it does. `self.nextToken` prints the next token and `SyntaxText(baseAddress: self.cursor.input.baseAddress, count: self.cursor.input.count)` prints the remaining source text thereafter. 

In my experience printing `self.lexemes` is useful to disambiguate where in the source file the current token is positioned, e.g. if there are multiple function declarations in the file and we’re currently at a `func` keyword.

If you want to inspect the next token in particular, you can always `po self.peek()`, which will print the next token similar to `self.currentToken`.

Reply:
> Like *Ptr

I think that line of reasoning works for Lexer.Cursor, but for the lexeme sequence I would very much expect to see the current token, the lookahead token, and then the rest of the buffer.

Reply:
How exactly would you expect those to be printed?

Reply:
Since this has sat long enough, I'll just ask for a delimiter between these two pieces to separate the head and tail of the sequence. I still think we ought to limit the amount of text returned in that tail (even lldb gives up after a while and gives you ...), but it's not as important.